### ISSUE TITLE
Should we use refresh=true on production? (Test is not suitable for 7.9)

### ISSUE THREAD

**Issue by bart2001** [lior-k/fast-elasticsearch-vector-scoring#61]
Hello, I would like to raise an issue about the test case.

According to `fast-elasticsearch-vector-scoring/src/test/java/com/liorkn/elasticsearch/PluginTest.java` file,
the test case uses `params.put("refresh", "true")` on data insert request, which makes vector scoring result works properly.
```java
public void test() throws Exception {
  final Map<String, String> params = new HashMap<>();
  params.put("refresh", "true");
```

However, in production case, we often do not use `refresh=true` option on data insertion.
If we do not use this option, the vector scoring result does workout properly. (such as same consine similarity score among result documents...)

I think one of two options should be taken as measure
1. Modify the test code (removing `refresh=true`) -> this may cause code level modification of vector scoring plugin
2. Mention in the document that `refresh=true` option must be provided on data insertion

**Comment by lior-k** [comment]
There is no need to use refresh=true in production.
This plugin does not require that. 
Refresh=true just means the inserted doc is quariable immediately after insertion, which is a must for the test.
 https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-refresh.html

**Comment by bart2001** [comment]
Hello, thank you for your reply.

The issue can be specified like this.

```
- Case 1. insert single data without `refresh=true` => vector scoring fail (cosine similarity)
- Case 2. insert singe data with `refresh=true` => vector scoring success (cosine similarity)
- Case 3. bulk insert with `refresh=true` => vector scoring fail (cosine similarity)
```
* `vector scoring fail` means there are same similarity scores among result documents, which does not guarantee correct ordering of result vectors

In summary, I think the problem is `refresh=true` is required to get "proper vector scoring result"...

This issue may be connected to compatibility with elasitcsearch core since the above cases all worked as `vector scoring success` on elasticsearch 7.5.2 version.

**Comment by lior-k** [comment]
right, thanks for clarifying. I was just able to recreate it in a test
Thanks for the update. I have no clue at the moment as of why this unexpected change in behaviour in this ES version
